x-pack/plugins/data_visualizer/public/application/data_comparison/data_comparison_view.tsx

```diff
@@ -0,0 +1,220 @@
+          )
+          .map((f) => ({
+            field: f.name,
+            type: getDataComparisonType(f.type),
+            displayName: f.displayName,
+          }))
+      );
+    }
+    setFetchIno({
+      fields: mergedFields,
+      currentDataView: dataView,
+      ...(windowParameters
+        ? {
+            timeRanges: {
+              reference: {
+                start: windowParameters.baselineMin,
+                end: windowParameters.baselineMax,
+              },
+              production: {
+                start: windowParameters.deviationMin,
+                end: windowParameters.deviationMax,
+              },
+            },
+          }
+        : {}),
+    });
+  }, [dataView, windowParameters]);
+  const result = useFetchDataComparisonResult({
+    ...fetchInfo,
+    lastRefresh,
+    randomSampler,
+    searchString,
+    searchQueryLanguage,
+    searchQuery,
+  });
+
+  const filteredData = useMemo(() => {
+    if (!result?.data) return [];
+
+    switch (showDataComparisonOnly) {
+      case true:
+        return result.data.filter((d) => d.driftDetected === true);
+      default:
+        return result.data;
+    }
+  }, [result.data, showDataComparisonOnly]);
+
+  const { onTableChange, pagination, sorting, setPageIndex } = useTableState<Feature>(
+    filteredData,
+    'driftDetected',
+    'desc'
+  );
+
+  const shouldRerunAnalysis = useMemo(
+    () =>
+      currentAnalysisWindowParameters !== undefined &&
+      !isEqual(currentAnalysisWindowParameters, windowParameters),
+    [currentAnalysisWindowParameters, windowParameters]
+  );
+
+  const onShowDataComparisonOnlyToggle = (e: EuiSwitchEvent) => {
+    setShowDataComparisonOnly(e.target.checked);
+    setPageIndex(0);
+  };
+
+  return windowParameters === undefined ? (
+    <EuiEmptyPrompt
+      color="subdued"
+      hasShadow={false}
+      hasBorder={false}
+      css={{ minWidth: '100%' }}
+      title={
+        <h2>
+          <FormattedMessage
+            id="xpack.dataVisualizer.dataComparison.emptyPromptTitle"
+            defaultMessage="Select a time range for reference and comparison data in the histogram chart to compare data distribution."
+          />
+        </h2>
+      }
+      titleSize="xs"
+      body={
+        <p>
+          <FormattedMessage
+            id="xpack.dataVisualizer.dataComparison.emptyPromptBody"
+            defaultMessage="The Data Comparison View compares the statistical properties of features in the 'reference' and 'comparison' data sets.
+"
+          />
+        </p>
+      }
+      data-test-subj="dataVisualizerNoWindowParametersEmptyPrompt"
+    />
+  ) : (
+    <div>
+      <ProgressControls
+        isBrushCleared={isBrushCleared}
+        onReset={onReset}
+        progress={result.loaded}
+        progressMessage={result.progressMessage ?? ''}
+        isRunning={result.loaded > 0 && result.loaded < 1}
+        onRefresh={updateFieldsAndTime}
+        onCancel={() => {}}
```

Comment:
Shouldn't this have a cancel action? Currently the cancel button doesn't seem to have any effect - for example here I've hit an error but hitting cancel has no effect visually.

<img width="1414" alt="image" src="https://github.com/elastic/kibana/assets/7405507/91d535fa-3f3d-477a-a9d4-fafa7f5cfd30">

Reply:
Fixed here [`895edd9` (#161365)](https://github.com/elastic/kibana/pull/161365/commits/895edd99dff23c8fefb2f57fbc566486c11f5b34). Canceling should now look like this:

https://github.com/elastic/kibana/assets/43350163/3d08f57a-2f72-4310-9cd9-2e8aec45cad8